### ISSUE TITLE
Harfbuzz's CMake config uses the wrong path

### ISSUE THREAD

**Issue by 2xsaiko** [NixOS/nixpkgs#180054]
### Describe the bug
Trying to use harfbuzz in a CMake project, CMake fails with the following:

```
CMake Error in CMakeLists.txt:
  Imported target "harfbuzz::harfbuzz" includes non-existent path

    "/nix/store/7ahz2g1i3m2pfk96fjl6j37ldcwlxpbg-harfbuzz-3.3.2-dev//nix/store/7ahz2g1i3m2pfk96fjl6j37ldcwlxpbg-harfbuzz-3.3.2-dev/include/harfbuzz"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.
```

That path comes from /nix/store/7ahz2g1i3m2pfk96fjl6j37ldcwlxpbg-harfbuzz-3.3.2-dev/lib/cmake/harfbuzz/harfbuzz-config.cmake, probably on line 25 which says `set(_harfbuzz_includedir "${prefix}//nix/store/7ahz2g1i3m2pfk96fjl6j37ldcwlxpbg-harfbuzz-3.3.2-dev/include")`.

### Steps To Reproduce
1. Make a CMake project that tries to link an executable to harfbuzz:
    ```
    find_package(harfbuzz REQUIRED)
    add_executable(program main.c)
    target_link_libraries(program harfbuzz::harfbuzz)
    ```
3. Try to configure project (`cmake .`)
4. See error above

### Expected behavior
The project configures successfully.

### Notify maintainers

@edolstra 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.49, NixOS, 22.11 (Raccoon), 22.11.20220627.020c740`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.9.1`
 - nixpkgs: `/etc/nix/inputs/nixpkgs`
```

**Comment by 2xsaiko** [comment]
I think I found a potential clean solution, this patch applies to harfbuzz. Meson has a path join operator, which I made it use instead of the plain string concat operation:

```patch
diff -Naur result/src/meson.build src/src/meson.build
--- a/src/meson.build   1970-01-01 01:00:01.000000000 +0100
+++ b/src/meson.build   2022-10-26 19:05:09.699922047 +0200
@@ -651,8 +651,8 @@
 have_gobject = conf.get('HAVE_GOBJECT', 0) == 1
 
 cmake_config = configuration_data()
-cmake_config.set('libdir', '${prefix}/@0@'.format(get_option('libdir')))
-cmake_config.set('includedir', '${prefix}/@0@'.format(get_option('includedir')))
+cmake_config.set('libdir', '${prefix}' / '@0@'.format(get_option('libdir')))
+cmake_config.set('includedir', '${prefix}' / '@0@'.format(get_option('includedir')))
 cmake_config.set('HB_LIBTOOL_VERSION_INFO', hb_libtool_version_info)
 cmake_config.set('have_gobject', '@0@'.format(have_gobject))
 configure_file(input: 'harfbuzz-config.cmake.in',
```

Going to see if I can upstream it.

**Comment by 2xsaiko** [comment]
While the last one didn't work, this one does but it's nowhere as clean. Oh well.

```patch
diff --git a/src/harfbuzz-config.cmake.in b/src/harfbuzz-config.cmake.in
index 304410d9b..43c10a0e2 100644
--- a/src/harfbuzz-config.cmake.in
+++ b/src/harfbuzz-config.cmake.in
@@ -24,6 +24,7 @@ unset(_harfbuzz_libdir_iter)
 # Get the include subdir.
 set(_harfbuzz_includedir "@includedir@")
 string(REPLACE "${_harfbuzz_remove_string}/" "" _harfbuzz_includedir "${_harfbuzz_includedir}")
+cmake_path(APPEND "${_harfbuzz_prefix}" "${_harfbuzz_includedir}" OUTPUT_VARIABLE _harfbuzz_includedir)
 
 # Extract version information from libtool.
 set(_harfbuzz_version_info "@HB_LIBTOOL_VERSION_INFO@")
@@ -48,18 +49,18 @@ endif ()
 # Add the libraries.
 add_library(harfbuzz::harfbuzz SHARED IMPORTED)
 set_target_properties(harfbuzz::harfbuzz PROPERTIES
-  INTERFACE_INCLUDE_DIRECTORIES "${_harfbuzz_prefix}/${_harfbuzz_includedir}/harfbuzz"
+  INTERFACE_INCLUDE_DIRECTORIES "${_harfbuzz_includedir}/harfbuzz"
   IMPORTED_LOCATION "${_harfbuzz_prefix}/${_harfbuzz_libdir}/libharfbuzz${_harfbuzz_lib_suffix}")
 
 add_library(harfbuzz::icu SHARED IMPORTED)
 set_target_properties(harfbuzz::icu PROPERTIES
-  INTERFACE_INCLUDE_DIRECTORIES "${_harfbuzz_prefix}/${_harfbuzz_includedir}/harfbuzz"
+  INTERFACE_INCLUDE_DIRECTORIES "${_harfbuzz_includedir}/harfbuzz"
   INTERFACE_LINK_LIBRARIES "harfbuzz::harfbuzz"
   IMPORTED_LOCATION "${_harfbuzz_prefix}/${_harfbuzz_libdir}/libharfbuzz-icu${_harfbuzz_lib_suffix}")
 
 add_library(harfbuzz::subset SHARED IMPORTED)
 set_target_properties(harfbuzz::subset PROPERTIES
-  INTERFACE_INCLUDE_DIRECTORIES "${_harfbuzz_prefix}/${_harfbuzz_includedir}/harfbuzz"
+  INTERFACE_INCLUDE_DIRECTORIES "${_harfbuzz_includedir}/harfbuzz"
   INTERFACE_LINK_LIBRARIES "harfbuzz::harfbuzz"
   IMPORTED_LOCATION "${_harfbuzz_prefix}/${_harfbuzz_libdir}/libharfbuzz-subset${_harfbuzz_lib_suffix}")
 
@@ -68,7 +69,7 @@ set(_harfbuzz_have_gobject "@have_gobject@")
 if (_harfbuzz_have_gobject)
   add_library(harfbuzz::gobject SHARED IMPORTED)
   set_target_properties(harfbuzz::gobject PROPERTIES
-    INTERFACE_INCLUDE_DIRECTORIES "${_harfbuzz_prefix}/${_harfbuzz_includedir}/harfbuzz"
+    INTERFACE_INCLUDE_DIRECTORIES "${_harfbuzz_includedir}/harfbuzz"
     INTERFACE_LINK_LIBRARIES "harfbuzz::harfbuzz"
     IMPORTED_LOCATION "${_harfbuzz_prefix}/${_harfbuzz_libdir}/libharfbuzz-gobject${_harfbuzz_lib_suffix}")
 endif ()
```

**Comment by 2xsaiko** [comment]
Fixed in harfbuzz 6.0.0.
